das/metrics.go

```diff
@@ -24,7 +25,10 @@ type metrics struct {
 	sampleTime    syncfloat64.Histogram
 	getHeaderTime syncfloat64.Histogram
 	newHead       syncint64.Counter
-	lastSampledTS int64
+	totalSampled  asyncint64.Gauge
```

Comment:
this looks like it's unused on the struct - why are you storing it? you only need it for the callback func

Reply:
will keep ftm to make use of dependency injection during test to inject a mock.

Reply:
thought I'd need it for testing, turns out not needed, and in either case now tests are removed so...